cmstestsuite/unit_tests/grading/__init__.py

```diff
@@ -0,0 +1,174 @@
+        # Translation is applied before formatting.
+        self.assertEqual(format_status_text(["A%s", "ASD"], self._tr), "EASD")
+        self.assertEqual(
+            format_status_text(["AAA %s\n%s", "AAA", "AE"], self._tr),
+            "EEE AAA\nAE")
+
+    def test_insuccess(self):
+        self.assertEqual(format_status_text(["%s"]), "N/A")
+        self.assertEqual(format_status_text(["%s"], self._tr), "N/E")
+
+
+class TestMergeEvaluationResults(unittest.TestCase):
+
+    @staticmethod
+    def _res(execution_time, execution_wall_clock_time, execution_memory,
+             exit_status, signal=None, filename=None):
+        r = {
+            "execution_time": execution_time,
+            "execution_wall_clock_time": execution_wall_clock_time,
+            "execution_memory": execution_memory,
+            "exit_status": exit_status,
+        }
+        if signal is not None:
+            r["signal"] = signal
+        if filename is not None:
+            r["filename"] = filename
+        return r
+
+    def test_success_status_ok(self):
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(1.0, 2.0, 300, Sandbox.EXIT_OK),
+                self._res(0.1, 0.2, 0.3, Sandbox.EXIT_OK)),
+            self._res(1.1, 2.0, 300.3, Sandbox.EXIT_OK))
+
+    def test_success_first_status_ok(self):
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_OK),
+                self._res(0, 0, 0, Sandbox.EXIT_TIMEOUT)),
+            self._res(0, 0, 0, Sandbox.EXIT_TIMEOUT))
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_OK),
+                self._res(0, 0, 0, Sandbox.EXIT_FILE_ACCESS, filename="asd")),
+            self._res(0, 0, 0, Sandbox.EXIT_FILE_ACCESS, filename="asd"))
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_OK),
+                self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="11")),
+            self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="11"))
+
+    def test_success_first_status_not_ok(self):
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_TIMEOUT),
+                self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="11")),
+            self._res(0, 0, 0, Sandbox.EXIT_TIMEOUT))
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_FILE_ACCESS, filename="asd"),
+                self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="11")),
+            self._res(0, 0, 0, Sandbox.EXIT_FILE_ACCESS, filename="asd"))
+        self.assertEqual(
+            merge_evaluation_results(
+                self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="9"),
+                self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="11")),
+            self._res(0, 0, 0, Sandbox.EXIT_SIGNAL, signal="9"))
```

Comment:
Maybe add one where the second status is ok?

Reply:
Done.